src/app/shared/state/workflow.service.ts

```diff
@@ -106,6 +106,16 @@ export class WorkflowService {
     }
   }
 
+  /**
+   * Updates an existing workflow by updating the current list of workflows
+   * and updating extendededWorkfllowService
+   * @param workflow the workflow to be updated
+   */
+  updateWorkflow(workflow: BioWorkflow | Service) {
+    this.upsertWorkflowToWorkflow(workflow);
```

Comment:
What happens when you remove this line (don't update the list of workflows)?

Reply:
Good catch; I think I'd started out with this one, and when it didn't work, added the other line, not noticing this isn't necessary.

In testing this, I noticed that the next line is causing the currently selected repo to change, without changing the url, which leads to confusion.

Reply:
Removed it and it works.